packages/flutter_tools/lib/src/build_system/targets/android.dart

```diff
@@ -122,3 +123,150 @@ class ReleaseAndroidApplication extends CopyFlutterAotBundle {
     AotAndroidAssetBundle(),
   ];
 }
+
+/// Generate an ELF binary from a dart kernel file in release mode.
+///
+/// This rule implementation outputs the generated so to a unique location
+/// based on the Android ABI. This allows concurrent invocations of gen_snapshot
+/// to run simultaneously.
+///
+/// The name of an instance of this rule would be 'android_aot_profile_android-x64'
+/// and is relied upon by flutter.gradle to match the correct rule.
+///
+/// It will produce an 'app.so` in the build directory under a folder named with
+/// the matching Android ABI.
+class AndroidAot extends AotElfBase {
+  /// Create an [AndroidAot] implementation for a given [targetPlatform] and [buildMode].
+  const AndroidAot(this.targetPlatform, this.buildMode);
+
+  /// The name of the produced Android ABI.
+  String get _androidAbiName {
+    return getNameForAndroidArch(
+      getAndroidArchForName(getNameForTargetPlatform(targetPlatform)));
+  }
+
+  @override
+  String get name => 'android_aot_${getNameForBuildMode(buildMode)}_'
+    '${getNameForTargetPlatform(targetPlatform)}';
+
+  /// The specific Android ABI we are building for.
+  final TargetPlatform targetPlatform;
+
+  /// The selected build mode.
+  ///
+  /// This is restricted to [BuildMode.profile] or [BuildMode.relese].
+  final BuildMode buildMode;
+
+  @override
+  List<Source> get inputs => <Source>[
+    const Source.pattern('{FLUTTER_ROOT}/packages/flutter_tools/lib/src/build_system/targets/dart.dart'),
```

Comment:
q: why is this one part of the inputs?

Reply:
Sort of a work around to allow running from source to handle source file changes to the tool. It doesn't add much/any overhead to the build so it is okay to leave in.